sdk/servicebus/azure_messaging_servicebus/src/error.rs

```diff
@@ -94,8 +94,10 @@ impl ServiceBusError {
     }
 
     /// Returns the source error, if any.
-    pub fn source(&self) -> Option<&ServiceBusError> {
-        self.source.as_deref()
+    pub fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
```

Comment:
Why have a `source()` function on `ServiceBusError` which is a different implementation from the trait version? The two appear to be basically identical.